client/client.py

```
@@ -94,6 +84,46 @@
+
+
+async def list_tools(server: MCPServerStreamableHTTP) -> Tuple[List[Any], Optional[httpx.Response]]:
+    try:
+        tools = await server.list_tools()
+    except httpx.HTTPStatusError as exc:
+        return [], exc.response
+    except ExceptionGroup as exc:
+        for inner in flatten_exceptions(exc):
+            if isinstance(inner, httpx.HTTPStatusError):
+                return [], inner.response
+        raise exc
```

Ruff (0.14.1)

Use `raise` without specifying exception name

Remove exception name

(TRY201)